### ISSUE TITLE
Have parser skip metrics if not all fields are available.

### ISSUE THREAD

**Issue by R-Wang97** [wavefrontHQ/wavefront-kubernetes-collector#13]
Currently, the parser will crash if any of the required fields for a certain metric are missing. Is it possible to have the parser just skip those metrics and log a warning?

**Comment by vikramraman** [comment]
@R-Wang97 Thanks for raising an issue! Any stack traces or specifics, so we can reproduce it?

**Comment by R-Wang97** [comment]
@vikramraman  So for example, when decoding container stats for a pod, our `rootfs` object is as follows:
```
"rootfs": {
       "time": null
}
```
This leads to the following stack trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xec5fe6]

goroutine 80 [running]:
github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary.(*summaryMetricsSource).decodeEphemeralStorageStatsForContainer(0xc000464380, 0xc00014afc0, 0xc00036
8320, 0xc000368370)
        /go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary/summary.go:257 +0x86
github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary.(*summaryMetricsSource).decodeContainerStats(0xc000464380, 0xc0005b3380, 0xc00059f958, 0x0, 0x1c0d90
0)
  	/go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary/summary.go:219 +0x4c3
github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary.(*summaryMetricsSource).decodePodStats(0xc000464380, 0xc0005b31a0, 0xc00059fab0, 0xc00059fae0)
        /go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary/summary.go:193 +0x8b9
github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary.(*summaryMetricsSource).decodeSummary(0xc000464380, 0xc0002f6360, 0x0)
        /go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary/summary.go:113 +0x336
github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary.(*summaryMetricsSource).ScrapeMetrics(0xc000464380, 0x0, 0xed3d96674, 0x1bee1c0, 0x0, 0xed3d966b0, 0
x1bee1c0, 0xc0003daee0, 0x1, 0x1)
        /go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/summary/summary.go:82 +0xe1
github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources.scrape(0x1215300, 0xc000464380, 0x0, 0xed3d96674, 0x1bee1c0, 0x0, 0xed3d966b0, 0x1bee1c0, 0x400, 0x1211a40,
...)
    	/go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/manager.go:182 +0x7d
github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources.(*sourceManager).ScrapeMetrics.func1(0xc000525f00, 0x1215300, 0xc000464380, 0xc0005587e0, 0x0, 0xed3d96674,
0x1bee1c0, 0x0, 0xed3d966b0, 0x1bee1c0, ...)
        /go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/manager.go:113 +0x193
created by github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources.(*sourceManager).ScrapeMetrics
        /go/src/github.com/wavefronthq/wavefront-kubernetes-collector/plugins/sources/manager.go:107 +0x69d
```

The wavefront collector was initialized with the following options:
```
/wavefront-collector --enable-discovery=false --source=kubernetes.summary_api:'' --sink=wavefront:?proxyAddress=wavefront-proxy.default.
svc.cluster.local:2878&clusterName=boba2&includeLabels=true --v=2
```

**Comment by vikramraman** [comment]
Thanks!

**Comment by vikramraman** [comment]
@R-Wang97 Fixed the issue: https://github.com/wavefrontHQ/wavefront-kubernetes-collector/commit/91cb12b3b1029c3d2a3fa05245b6d7cd90a9f6ba

Would you mind trying the `0.9.6-rc1` image and let us know if that works?: https://hub.docker.com/r/wavefronthq/wavefront-kubernetes-collector/tags

**Comment by vikramraman** [comment]
Appears all the other cases directly delegate to `addIntMetric` or `addLabeledIntMetric` that already checks for nil values:

https://github.com/wavefrontHQ/wavefront-kubernetes-collector/blob/73c636d1112d71e0591a1bb2cbee10008320bae4/plugins/sources/summary/summary.go#L360

**Comment by vikramraman** [comment]
Released `0.9.6`. You can use that or `latest`. Thanks for reporting the issue!
